### PR TITLE
Documentation/proposals: Add a proposal for refresh token revocation.

### PR DESCRIPTION
**Pull request by rithujohn191**
Outline the design for handling refresh token revocation in Dex v2.

### CODE REVIEW

File: Documentation/proposals/token-revocation.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+//
+// Will have to handle garbage collection i.e. if no refresh token exists for a user,
+// this object must be cleaned up.
+type OfflineSession struct {
+        // UserID of an end user who has logged in to the server.
+        UserID        string
+        // The ID of the connector used to login the user.
+        ConnID     string
+        // List of pointers to RefreshTokens issued for SessionID
+        Refresh         []*RefreshToken
```
Comment (ericchiang) [inline Documentation/proposals/token-revocation.md]: "RefreshToken" should only contain metadata about the refresh token. It's not the actual refresh token type, right?

Comment (rithujohn191) [inline Documentation/proposals/token-revocation.md]: should I clarify this to be internal.RefreshToken?

Comment (ericchiang) [inline Documentation/proposals/token-revocation.md]: We should probably just create a `RefresTokenhRef` object.

File: Documentation/proposals/token-revocation.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+		* Create an OfflineSession for the user and add the new `RefreshToken` pointer to
+		  the `Refresh` list.
+                
+2. Refresh token rotation:
+   There will be no change to this codeflow. When the client refreshes a refresh token, the `TokenID`
+   still remains intact and only the `RefreshToken` obj gets updated with a new nonce. We do not need
+   any additional checks in the OfflineSession objects as the `RefreshToken` pointers still remain intact.
+
+3. User revokes a refresh token (New functionality):
+   A user that has been authenticated via a connector will have the ability to revoke their refresh tokens.
```
Comment (sym3tri) [inline Documentation/proposals/token-revocation.md]: How is authentication implemented? A valid JWT for this dex instance (iss) with a matching user ID?

Comment (ericchiang) [inline Documentation/proposals/token-revocation.md]: Oh, we should remove this. Our API doesn't do authentication or authorization. We'd want to leave it to another app.

Comment (rithujohn191) [inline Documentation/proposals/token-revocation.md]: Oh I see how that can be misleading. I will clarify that an external app will have to handle the authentication.

**Pull request merged**